qiskit/opflow/list_ops/tensored_op.py

```diff
@@ -116,3 +116,11 @@ def to_circuit(self) -> QuantumCircuit:
             "Conversion to_circuit supported only for operators, where a single "
             "underlying circuit can be produced."
         )
+
+    def to_matrix(self, massive: bool = False) -> np.ndarray:
+        OperatorBase._check_massive("to_matrix", True, self.num_qubits, massive)
+
+        mat = self.coeff * reduce(
+            np.kron, [np.asarray(op.to_matrix(massive=massive)) for op in self.oplist]
+        )
```

Comment:
If you've already checked that the output of the tensor product is massive, maybe you don't need to bother checking the internal matrices?

Reply:
Good point, I removed the second check in e6ab027